cloudify_cli/commands/executions.py

```diff
@@ -526,3 +532,318 @@ def delete(logger, client, tenant_name, to_datetime, before, keep_last,
+        stop_on_fail=stop_on_fail)
+
+    logger.info('Execution schedule created successfully')
+
+
+@schedule.command(name='update',
+                  short_help='Update a workflow execution schedule')
+@cfy.argument('name')
+@cfy.options.common_options
+@cfy.options.from_datetime(required=False, help="Time to schedule from.")
+@cfy.options.to_datetime(required=False, help="Time to schedule until.")
+@cfy.options.frequency
+@cfy.options.count
+@cfy.options.weekdays
+@cfy.options.rrule
+@cfy.options.slip
+@click.option(
+    '--stop-on-fail/--continue-on-fail',
+    required=False,
+    help=helptexts.SCHEDULE_STOP_ON_FAIL
+)
+@cfy.assert_manager_active()
+@cfy.options.tenant_name(required=False, resource_name_for_help='execution')
+@cfy.pass_client()
+@cfy.pass_logger
+def schedule_update(name,
+                    from_datetime,
+                    to_datetime,
+                    freq,
+                    count,
+                    weekdays,
+                    rrule,
+                    slip,
+                    stop_on_fail,
+                    tenant_name,
+                    client,
+                    logger):
+    """Update an existing schedule for a workflow execution
+    """
+    utils.explicit_tenant_name_message(tenant_name, logger)
+    logger.info('Updating execution schedule {0}...'.format(name))
+    client.execution_schedules.update(
+        name,
+        since=from_datetime,
+        until=to_datetime,
+        frequency=freq,
+        count=count,
+        weekdays=weekdays,
+        rrule=rrule,
+        slip=slip,
+        stop_on_fail=stop_on_fail)
+
+    logger.info('Execution schedule updated successfully')
+
+
+@schedule.command(name='preview',
+                  short_help='Preview the next running dates for a schedule. '
+                             'Works either with an existing schedule or with '
+                             'a set of provided scheduling parameters')
+@cfy.options.common_options
+@click.option('-n', '--name', required=False,
+              help="Name of an existing execution schedule.\nMutually "
+                   "exclusive with other options.")
+@cfy.options.from_datetime(required=False, help="Time to schedule from.")
+@cfy.options.to_datetime(required=False, help="Time to schedule until.")
+@cfy.options.frequency
+@cfy.options.count
+@cfy.options.weekdays
+@cfy.options.rrule
+@cfy.options.pagination_size
+@cfy.assert_manager_active()
+@cfy.options.tenant_name(required=False, resource_name_for_help='execution')
+@cfy.pass_client()
+@cfy.pass_logger
+def schedule_preview(name,
+                     from_datetime,
+                     to_datetime,
+                     freq,
+                     count,
+                     weekdays,
+                     rrule,
+                     tenant_name,
+                     pagination_size,
+                     client,
+                     logger):
+    """Preview the next running dates for a schedule. Works either with an
```

Comment:
it would be good if we were able to push the computation to the server-side, beause now we have two places that do this, and good luck making sure they're the same!

also this requires us to have dateutil and it would be good if we didnt have to add new dependencies to the cli